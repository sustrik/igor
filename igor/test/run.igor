package main

func main() {
    var i, j int
    var n nursery
    run(n, foo(i, j))
    n.Close()
}

func foo(i, j int) error {return nil}