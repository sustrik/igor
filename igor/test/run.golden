package main

import __context "context"
import __igor "github.com/sustrik/igor/lib/igor"

func __main(__ctx __context.Context) {
	var i, j int
	var n __igor.Nursery
	{
		__nursery := n
		__nursery.Start__()
		go func() {
			__err := foo(__nursery.Context__(), i, j)
			__nursery.Stop__(__err)
		}()
	}
	n.Close(__ctx)
}

func foo(__ctx __context.Context, i, j int) error { return nil }
func main() {
	__main(__context.Background())
}
